### PR TITLE
asm: remove integration tests with golang-asm

### PR DESCRIPTION
**Pull request by mathetake**

### CODE REVIEW

**Comment by mathetake** [comment]
lol 10k 😄

**Comment by mathetake** [comment]
ok less than 10k

File: internal/asm/amd64/impl.go [hunk 1]
```diff
@@ -10,148 +10,146 @@ import (
 	"github.com/tetratelabs/wazero/internal/asm"
```
Comment (mathetake) [inline internal/asm/amd64/impl.go:10]: all the change in this file is only un-exporting so nothing fun

**Review by codefromthecrypt — APPROVED** [review]
cool, so what's left on golang-asm removal?

**Comment by mathetake** [comment]
nothing left! removed completely... will change the title

**Pull request merged**